### PR TITLE
Add replyto msg

### PR DESCRIPTION
**Pull request by guste-github**
Add replyTo for writing and parsing msg files.

### CODE REVIEW

File: src/main/java/ch/astorm/jotlmsg/OutlookMessage.java [hunk 1]
```diff
@@ -573,7 +597,7 @@ public void writeTo(OutputStream outputStream) throws IOException {
                 attachStorage.setProperty(new PropertyValue(MAPIProperty.ATTACH_FILENAME, FLAG_READABLE | FLAG_WRITEABLE, StringUtil.getToUnicodeLE(name))); 
                 attachStorage.setProperty(new PropertyValue(MAPIProperty.ATTACH_LONG_FILENAME, FLAG_READABLE | FLAG_WRITEABLE, StringUtil.getToUnicodeLE(name))); 
             }
-            if(mimeType!=null) { attachStorage.setProperty(new PropertyValue(MAPIProperty.ATTACH_MIME_TAG, FLAG_READABLE | FLAG_WRITEABLE, StringUtil.getToUnicodeLE(name))); }
+            if(mimeType!=null) { attachStorage.setProperty(new PropertyValue(MAPIProperty.ATTACH_MIME_TAG, FLAG_READABLE | FLAG_WRITEABLE, StringUtil.getToUnicodeLE(mimeType))); }
```
Comment (ctabin) [inline src/main/java/ch/astorm/jotlmsg/OutlookMessage.java:600]: Mmmh that looks like a bug that should be put it another MR isn't it ? :thinking:

File: src/main/java/ch/astorm/jotlmsg/OutlookMessage.java [hunk 2]
```diff
@@ -635,7 +659,25 @@ protected void parseFrom(MAPIMessage mapiMessage) throws ChunkNotFoundException
...
+        if(replyEntriesChunks!=null && !replyEntriesChunks.isEmpty()) {
+            for(Chunk chunk : replyEntriesChunks) {
+                if(chunk instanceof ByteChunk) {
+                    ByteChunk bc = (ByteChunk)chunk;
+                    replyToRecipentBytes = bc.getValue();
+                }
+            }        
+        }
+        if(replyToRecipentBytes!=null) {
+            FlatEntryListStructure<OneOffEntryIDStructure> fels = new FlatEntryListStructure<OneOffEntryIDStructure>(OneOffEntryIDStructure.class, replyToRecipentBytes);
```
Comment (ctabin) [inline src/main/java/ch/astorm/jotlmsg/OutlookMessage.java:674]: Use diamond operator like above.

File: src/test/java/ch/astorm/jotlmsg/FlatEntryListStructureTest.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
+package ch.astorm.jotlmsg;
```
Comment (ctabin) [inline src/test/java/ch/astorm/jotlmsg/FlatEntryListStructureTest.java:1]: Move to `ch.astorm.jotlmsg.io`.

File: src/test/java/ch/astorm/jotlmsg/OneOffEntryIDStructureTest.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
+package ch.astorm.jotlmsg;
```
Comment (ctabin) [inline src/test/java/ch/astorm/jotlmsg/OneOffEntryIDStructureTest.java:1]: Move to `ch.astorm.jotlmsg.io`

**Review by ctabin — CHANGES_REQUESTED** [review]
@guste-github Tanks for your PR, it's awesome work ! I'm surprised to see so much code just to handle a `reply-to` feature :thinking: Can you maybe give me some deeper explanations about it ? Do you have a link to a specification ?

I did a first review pass to check the changes and make some general feedback. Can you please also cleanup the javadoc of your classes to remove useless flags and parameters ? Also I think that your feature is worth an update to the README.

**Comment by guste-github** [comment]
@ctabin Thanks a lot for your feedback! There are two relevant MAPIProperties which have to be used together.

MAPIProperty.REPLY_RECIPIENT_NAMES
MAPIProperty.REPLY_RECIPIENT_ENTRIES

2.2.1.43 PidTagReplyRecipientEntries Property
https://docs.microsoft.com/en-us/openspecs/exchange_server_protocols/ms-oxomsg/8f99a22f-3918-44ec-a97c-c31b4aecf05b

It's easy to fill data regarding the REPLY_RECIPIENT_NAMES property because it's a simple unicode string. 
The data regarding REPLY_RECIPIENT_ENTRIES is organized like old fashion c structs with length coding and alignment requirements.
But at the end it's a number of One-Off EntryID Structures.

[MS-OXCDATA] 2.3.3 FlatEntryList Structure
https://docs.microsoft.com/en-us/openspecs/exchange_server_protocols/ms-oxcdata/3d7e620c-808e-431f-9dc4-28eaf4066c23

[MS-OXCDATA] 2.3.2 FlatEntry Structure
https://docs.microsoft.com/en-us/openspecs/exchange_server_protocols/ms-oxcdata/3677fbcf-09a0-4377-a699-8ff4dc0a143a

[MS-OXCDATA] 2.2.5.1 One-Off EntryID Structure
https://docs.microsoft.com/en-us/openspecs/exchange_server_protocols/ms-oxcdata/b32d23af-85f6-4e92-8387-53a1950ae7ba

File: pom.xml [hunk 5]
```diff
@@ -87,6 +87,8 @@
         <configuration>
           <source>1.8</source>
           <target>1.8</target>
+          <!-- Replace source and target tag with release tag for compilation with JDK > 1.8 -->
+          <!-- <release>8</release> -->
```
[the remark this answers is not in the document]
Comment (guste-github) [reply pom.xml:91]: No problem. I will remove it.

File: src/main/java/ch/astorm/jotlmsg/FlatEntryListStructure.java [hunk 6]
```diff
@@ -0,0 +1,147 @@
+package ch.astorm.jotlmsg;
```
[the remark this answers is not in the document]
Comment (guste-github) [reply src/main/java/ch/astorm/jotlmsg/FlatEntryListStructure.java:1]: OK.

File: src/main/java/ch/astorm/jotlmsg/FlatEntryStructureFactory.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
+package ch.astorm.jotlmsg;
```
[the remark this answers is not in the document]
Comment (guste-github) [reply src/main/java/ch/astorm/jotlmsg/FlatEntryStructureFactory.java:1]: OK.

File: src/main/java/ch/astorm/jotlmsg/FlatEntryStructure.java [hunk 8]
```diff
@@ -0,0 +1,62 @@
+package ch.astorm.jotlmsg;
```
[the remark this answers is not in the document]
Comment (guste-github) [reply src/main/java/ch/astorm/jotlmsg/FlatEntryStructure.java:1]: OK.

File: src/main/java/ch/astorm/jotlmsg/OneOffEntryIDStructure.java [hunk 9]
```diff
@@ -0,0 +1,121 @@
+package ch.astorm.jotlmsg;
```
[the remark this answers is not in the document]
Comment (guste-github) [reply src/main/java/ch/astorm/jotlmsg/OneOffEntryIDStructure.java:1]: OK.

File: src/main/java/ch/astorm/jotlmsg/OutlookMessage.java [hunk 10]
```diff
@@ -517,6 +518,29 @@ public void writeTo(OutputStream outputStream) throws IOException {
             topLevelChunk.setProperty(new PropertyValue(MAPIProperty.SENDER_EMAIL_ADDRESS, FLAG_READABLE | FLAG_WRITEABLE, StringUtil.getToUnicodeLE(from))); 
             topLevelChunk.setProperty(new PropertyValue(MAPIProperty.SENDER_NAME, FLAG_READABLE | FLAG_WRITEABLE, StringUtil.getToUnicodeLE(from))); 
         }
+
+        //creates the reply recipients
+        if(replyToRecipents!=null) {
+            FlatEntryListStructure<OneOffEntryIDStructure> fels = new FlatEntryListStructure<OneOffEntryIDStructure>();
```
[the remark this answers is not in the document]
Comment (guste-github) [reply src/main/java/ch/astorm/jotlmsg/OutlookMessage.java:524]: I agree.

File: src/main/java/ch/astorm/jotlmsg/OutlookMessage.java (hunk 2 above)
Comment (guste-github) [reply src/main/java/ch/astorm/jotlmsg/OutlookMessage.java:674]: I agree.

File: src/test/java/ch/astorm/jotlmsg/FlatEntryListStructureTest.java (hunk 3 above)
Comment (guste-github) [reply src/test/java/ch/astorm/jotlmsg/FlatEntryListStructureTest.java:1]: OK.

File: src/test/java/ch/astorm/jotlmsg/OneOffEntryIDStructureTest.java (hunk 4 above)
Comment (guste-github) [reply src/test/java/ch/astorm/jotlmsg/OneOffEntryIDStructureTest.java:1]: OK.

File: src/main/java/ch/astorm/jotlmsg/OutlookMessage.java (hunk 1 above)
Comment (guste-github) [reply src/main/java/ch/astorm/jotlmsg/OutlookMessage.java:600]: Yes, seems to be a very simple copy paste error.

File: pom.xml [hunk 11]
```diff
@@ -208,5 +211,11 @@
       <version>5.6.2</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+        <groupId>org.junit.jupiter</groupId>
+        <artifactId>junit-jupiter-engine</artifactId>
+        <version>5.6.2</version>
+        <scope>test</scope>        
+    </dependency>
```
[the remark this answers is not in the document]
Comment (guste-github) [reply pom.xml:219]: No problem.

File: src/main/java/ch/astorm/jotlmsg/io/FlatEntryListStructure.java [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+ * Represents an [MS-OXCDATA] 2.3.3 FlatEntryList Structure in Java.
+ * 
+ * @author Guido Stein
+ *
+ */
+public class FlatEntryListStructure<T extends FlatEntryStructure> implements Iterable<T> {
+    private long count; // Number of FlatEntryStructures.
+    private long size; // Number of bytes of all FlatEntryStructures.
+    private List<T> flatEntryStructures = new ArrayList<T>();
+    private FlatEntryStructureFactory<T> fesf = new FlatEntryStructureFactory<T>();
```
Comment (ctabin) [inline src/main/java/ch/astorm/jotlmsg/io/FlatEntryListStructure.java:19]: Use the diamond operator => `new ArrayList<>()`.

File: src/main/java/ch/astorm/jotlmsg/io/FlatEntryListStructure.java [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+        size = totalSize;
+        count = (long) flatEntryStructures.size();
+    }
+
+    /**
+     * Constructor, creates an empty java list representation.
+     * 
+     */
+    public FlatEntryListStructure() {
+    };
```
Comment (ctabin) [inline src/main/java/ch/astorm/jotlmsg/io/FlatEntryListStructure.java:84]: The whole constructor can be entirely removed.

File: src/main/java/ch/astorm/jotlmsg/io/FlatEntryListStructure.java [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+            bf.position((bf.position() + 4) & ~3);
+        }
+        size = totalSize - 8;
+        return bf.array();
+    }
+
+    /**
+     * Iterator.
+     */
+    public Iterator<T> iterator() {
```
Comment (ctabin) [inline src/main/java/ch/astorm/jotlmsg/io/FlatEntryListStructure.java:144]: Note sure about this javadoc comment :thinking: Maybe just remove it ? Also, it seems that `@Override` is missing.

File: src/main/java/ch/astorm/jotlmsg/io/FlatEntryStructure.java [hunk 15]
```diff
@@ -0,0 +1,62 @@
+package ch.astorm.jotlmsg.io;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Represents an [MS-OXCDATA] 2.3.2 FlatEntry Structure in Java.
+ * 
+ * @author Guido Stein
+ *
```
Comment (ctabin) [inline src/main/java/ch/astorm/jotlmsg/io/FlatEntryStructure.java:9]: Please remove this empty line.

File: src/main/java/ch/astorm/jotlmsg/io/FlatEntryStructure.java [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+    public void setEntryID(byte[] entryID) {
+        this.entryID = entryID;
+        this.size = entryID.length;
+    }
+
+    /**
+     * Default constructor
+     */
+    public FlatEntryStructure() {
+    };
```
Comment (ctabin) [inline src/main/java/ch/astorm/jotlmsg/io/FlatEntryStructure.java:47]: This constructor can be entirely removed.

File: src/main/java/ch/astorm/jotlmsg/io/FlatEntryStructureFactory.java [hunk 17]
```diff
@@ -0,0 +1,23 @@
+package ch.astorm.jotlmsg.io;
+
+import java.lang.reflect.InvocationTargetException;
+import java.nio.ByteBuffer;
+
+/**
+ * Creates FlatEntryStructures.
+ * 
+ * @author Guido Stein
+ *
```
Comment (ctabin) [inline src/main/java/ch/astorm/jotlmsg/io/FlatEntryStructureFactory.java:10]: Emplty line to be removed.

File: src/main/java/ch/astorm/jotlmsg/io/FlatEntryStructureFactory.java [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+ */
+public class FlatEntryStructureFactory<T extends FlatEntryStructure> {
```
Comment (ctabin) [inline src/main/java/ch/astorm/jotlmsg/io/FlatEntryStructureFactory.java:12]: Is this class really relevant ? Why not simply do `new MyFlatEntryStructure(bf)` where needed ?

File: src/main/java/ch/astorm/jotlmsg/io/FlatEntryStructureFactory.java [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+    public T createFlatEntryStructure(Class<T> type, ByteBuffer bf) {
+        T fes = null;
+        try {
+            fes = type.getDeclaredConstructor(new Class[] { ByteBuffer.class }).newInstance(bf);
+        } catch (InstantiationException | IllegalAccessException | IllegalArgumentException | InvocationTargetException
+                | NoSuchMethodException | SecurityException e) {
+            e.printStackTrace();
```
Comment (ctabin) [inline src/main/java/ch/astorm/jotlmsg/io/FlatEntryStructureFactory.java:19]: Also here, letting the stacktrace just be printed would lead to and unseen problem if there is an exception... Add a `throw new RuntimeException(e)` instead.

File: src/main/java/ch/astorm/jotlmsg/io/OneOffEntryIDStructure.java [hunk 20]
```diff
@@ -0,0 +1,121 @@
...
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+import org.apache.poi.util.StringUtil;
+
+/**
+ * Represents an [MS-OXCDATA] 2.2.5.1 One-Off EntryID Structure in Java.
+ * 
+ * @author Guido Stein
+ *
```
Comment (ctabin) [inline src/main/java/ch/astorm/jotlmsg/io/OneOffEntryIDStructure.java:12]: Empty line to be removed.

File: src/test/java/ch/astorm/jotlmsg/OutlookMessageMSGTest.java [hunk 21]
```diff
@@ -68,6 +71,11 @@ public void testMessageWithAttachment2() throws Exception {
         message.addAttachment("message3.txt", "text/html", new ByteArrayInputStream("<html><body>Some html page</body></html>".getBytes(StandardCharsets.UTF_8)));
         
         testMessage(message);
+        
+//        // There was a bug regarding the mimetype, so just generate a new reference file.
+//        File msgFile = new File("src/test/resources/ch/astorm/jotlmsg/generated/with-attachments-2.msg");        
+//        message.writeTo(msgFile);
```
Comment (ctabin) [inline src/test/java/ch/astorm/jotlmsg/OutlookMessageMSGTest.java:77]: Since this has been fixed, could you please rebase and regenerate it ?

File: src/test/java/ch/astorm/jotlmsg/OutlookMessageMSGTest.java [hunk 22]
```diff
@@ -128,8 +136,19 @@ private void compareMessage(OutlookMessage source, OutlookMessage other) throws
...
+            for(int i=0 ; i<srcReplyToRecipients.size() ; ++i) {
+            	String srcReplyToRecipient = srcReplyToRecipients.get(i);
+            	String parsedReplyToRecipient = parsedReplyToRecipients.get(i);
+                assertEquals(srcReplyToRecipient, parsedReplyToRecipient);
+            }
+        }
+        
         List<OutlookMessageRecipient> srcRecipients = source.getAllRecipients();
-        List<OutlookMessageRecipient> parsedRecipients = source.getAllRecipients();
+        List<OutlookMessageRecipient> parsedRecipients = other.getAllRecipients();
```
Comment (ctabin) [inline src/test/java/ch/astorm/jotlmsg/OutlookMessageMSGTest.java:151]: Damn... well seen ! This is fixed in #61, please rebase.

File: src/test/java/ch/astorm/jotlmsg/OutlookMessageMSGTest.java [hunk 23]
```diff
@@ -175,6 +193,24 @@ public void addManyAttachments() throws Exception {
         message.setSubject("betreff");
         message.setPlainTextBody("content");
 
+//        // There was a bug regarding the mimetype, so just generate a new reference file.
+//        File msgFile = new File("src/test/resources/ch/astorm/jotlmsg/generated/many-attachments.msg");        
+//        message.writeTo(msgFile);
```
Comment (ctabin) [inline src/test/java/ch/astorm/jotlmsg/OutlookMessageMSGTest.java:198]: This could also be uncommented.

File: src/test/java/ch/astorm/jotlmsg/io/FlatEntryListStructureTest.java [hunk 24]
```diff
@@ -0,0 +1,111 @@
...
+import org.apache.poi.hsmf.datatypes.ByteChunk;
+import org.apache.poi.hsmf.datatypes.Chunk;
+import org.apache.poi.hsmf.datatypes.MAPIProperty;
+import org.junit.jupiter.api.Test;
+
+public class FlatEntryListStructureTest {
+
+    @Test
+    public void testFlatEntryListStructure1() throws IOException {
+        InputStream msg = OneOffEntryIDStructureTest.class.getResourceAsStream("../msoutlook/replyto.msg");
```
Comment (ctabin) [inline src/test/java/ch/astorm/jotlmsg/io/FlatEntryListStructureTest.java:22]: Use try-with-resources.

File: src/test/java/ch/astorm/jotlmsg/io/FlatEntryListStructureTest.java [hunk 25]
```diff
@@ -0,0 +1,111 @@
...
+            assertArrayEquals(ooe1.getEntryID(), ooe2.getEntryID());
+        }
+        
+        mapiMessage.close();
+        msg.close();
+    }
+	
+    @Test
+    public void testFlatEntryListStructure2() throws IOException {
+        InputStream msg = FlatEntryListStructureTest.class.getResourceAsStream("../msoutlook/replyto.msg");
```
Comment (ctabin) [inline src/test/java/ch/astorm/jotlmsg/io/FlatEntryListStructureTest.java:61]: Use try-with-resources.

File: src/test/java/ch/astorm/jotlmsg/io/OneOffEntryIDStructureTest.java [hunk 26]
```diff
@@ -0,0 +1,132 @@
...
+import org.apache.poi.hsmf.datatypes.ByteChunk;
+import org.apache.poi.hsmf.datatypes.Chunk;
+import org.apache.poi.hsmf.datatypes.MAPIProperty;
+import org.junit.jupiter.api.Test;
+
+public class OneOffEntryIDStructureTest {
+
+    @Test
+    public void testOneOffEntryIDStructure1() throws IOException {
+        InputStream msg = OneOffEntryIDStructureTest.class.getResourceAsStream("../msoutlook/replyto.msg");
```
Comment (ctabin) [inline src/test/java/ch/astorm/jotlmsg/io/OneOffEntryIDStructureTest.java:22]: Use try-with-resources.

File: src/test/java/ch/astorm/jotlmsg/io/OneOffEntryIDStructureTest.java [hunk 27]
```diff
@@ -0,0 +1,132 @@
...
+        assertEquals("reply2@test.com", ooe2.getDisplayName());
+        assertEquals("reply2@test.com", ooe2.getEmailAddress());
+        
+        mapiMessage.close();
+        msg.close();
+    }
+
+    @Test
+    public void testOneOffEntryIDStructure2() throws IOException {
+        InputStream msg = OneOffEntryIDStructureTest.class.getResourceAsStream("../msoutlook/replyto.msg");
```
Comment (ctabin) [inline src/test/java/ch/astorm/jotlmsg/io/OneOffEntryIDStructureTest.java:60]: Use try-with-resources.

**Review by ctabin — CHANGES_REQUESTED** [review]
Sorry for coming back so late. Thanks for the work :thumbsup: I've left some comments.

File: src/main/java/ch/astorm/jotlmsg/OutlookMessage.java [hunk 28]
```diff
@@ -635,7 +661,25 @@ protected void parseFrom(MAPIMessage mapiMessage) throws ChunkNotFoundException
      * @throws ChunkNotFoundException If some data is not found in the {@code mapiMessage}.
      */
     protected void parseReplyTo(MAPIMessage mapiMessage) throws ChunkNotFoundException {
-        //TODO
+        // Note: There should be also a responding MAPIProperty.REPLY_RECIPIENT_NAMES entry.
+    	byte[] replyToRecipentBytes = null;
+        List<Chunk> replyEntriesChunks = mapiMessage.getMainChunks().getAll().get(MAPIProperty.REPLY_RECIPIENT_ENTRIES);
```
[the remark this answers is not in the document]
Comment (guste-github) [reply src/main/java/ch/astorm/jotlmsg/OutlookMessage.java:666]: Done.

File: src/main/java/ch/astorm/jotlmsg/OutlookMessage.java [hunk 29]
```diff
@@ -635,7 +661,25 @@ protected void parseFrom(MAPIMessage mapiMessage) throws ChunkNotFoundException
...
+                    ByteChunk bc = (ByteChunk)chunk;
+                    replyToRecipentBytes = bc.getValue();
+                }
+            }        
+        }
+        if(replyToRecipentBytes!=null) {
+            FlatEntryListStructure<OneOffEntryIDStructure> fels = new FlatEntryListStructure<>(OneOffEntryIDStructure.class, replyToRecipentBytes);
+            List<String> replyToRecipents = new ArrayList<String>();
+            for(OneOffEntryIDStructure ooes : fels) {
+            	replyToRecipents.add(ooes.getEmailAddress());
```
[the remark this answers is not in the document]
Comment (guste-github) [reply src/main/java/ch/astorm/jotlmsg/OutlookMessage.java:679]: Done.

File: src/main/java/ch/astorm/jotlmsg/io/FlatEntryListStructure.java (hunk 14 above)
Comment (guste-github) [reply src/main/java/ch/astorm/jotlmsg/io/FlatEntryListStructure.java:144]: I think so too. The comment is useless and I will add the missing override anotation.

File: src/test/java/ch/astorm/jotlmsg/OutlookMessageMSGTest.java (hunk 21 above)
Comment (guste-github) [reply src/test/java/ch/astorm/jotlmsg/OutlookMessageMSGTest.java:77]: Yes, sure.

File: src/test/java/ch/astorm/jotlmsg/io/FlatEntryListStructureTest.java (hunk 24 above)
Comment (guste-github) [reply src/test/java/ch/astorm/jotlmsg/io/FlatEntryListStructureTest.java:22]: Introduced in Java 7 and I have never used it before... Nice.

File: src/main/java/ch/astorm/jotlmsg/io/FlatEntryListStructure.java (hunk 12 above)
Comment (guste-github) [reply src/main/java/ch/astorm/jotlmsg/io/FlatEntryListStructure.java:19]: OK.

File: src/main/java/ch/astorm/jotlmsg/io/FlatEntryStructureFactory.java (hunk 18 above)
Comment (guste-github) [reply src/main/java/ch/astorm/jotlmsg/io/FlatEntryStructureFactory.java:12]: The replyto data is stored within the msg file like an array of structs in C. So FlatEntryListStructure and FlatEntryStructures provides some basis functionality like memory alignment and access to the binary data. OneOffEntryIDStructure is a special kind of FlatEntryStructure to interpret the data we need. But FlatEntryListStructure and FlatEntryStructures shouldn't work only with OneOffEntryIDStructure. So FlatEntryStructureFactory contains some generics to instantiate subclasses of FlatEntryStructure.

File: src/main/java/ch/astorm/jotlmsg/io/FlatEntryStructureFactory.java (hunk 19 above)
Comment (guste-github) [reply src/main/java/ch/astorm/jotlmsg/io/FlatEntryStructureFactory.java:19]: You are absolutely right. Bad idea! I will replace it. 👍

File: src/main/java/ch/astorm/jotlmsg/io/FlatEntryStructure.java (hunk 16 above)
Comment (guste-github) [reply src/main/java/ch/astorm/jotlmsg/io/FlatEntryStructure.java:47]: I hava tried to remove it but it seems that the generic stuff doesn't work without it.

File: src/main/java/ch/astorm/jotlmsg/io/FlatEntryListStructure.java (hunk 13 above)
Comment (guste-github) [reply src/main/java/ch/astorm/jotlmsg/io/FlatEntryListStructure.java:84]: I hava tried to remove it but it seems that the generic stuff doesn't work without it.

File: src/main/java/ch/astorm/jotlmsg/io/FlatEntryStructureFactory.java (hunk 18 above)
Comment (ctabin) [reply src/main/java/ch/astorm/jotlmsg/io/FlatEntryStructureFactory.java:11]: OK thanks for the details. Let's keep it !

**Comment by ctabin** [comment]
@guste-github I'm really sorry to come back this late ! I've been very busy these times, but did not forget about your PR.

Your changes are good for me, let's merge them :pray: :thumbsup: :thumbsup: :tada: :tada:
